packages/gatsby-source-contentful/src/fetch.js

```diff
@@ -4,14 +4,70 @@ const chalk = require(`chalk`)
 const { formatPluginOptionsForCLI } = require(`./plugin-options`)
 const { CODES } = require(`./report`)
 
+const createContentfulErrorMessage = e => {
+  if (typeof e === `string`) {
+    return e
+  }
+
+  // If we get a response, use this as basis to create the error message
+  // Contentful JS SDK tends to give errors with different structures:
+  // https://github.com/contentful/contentful.js/blob/b67b77ac8c919c4ec39203f8cac2043854ab0014/lib/create-contentful-api.js#L89-L99
+  if (e.response) {
+    e = { ...e, ...e.response }
+
+    if (typeof e.response.data === `string`) {
+      try {
+        const data = JSON.parse(e.response.data)
+        if (data && typeof data === `object`) {
+          e = { ...e, ...data }
+        }
+      } catch (err) {
+        e.message = `Unable to extract API error data from:\n${e.response.data}`
+      }
+    } else if (typeof e.response.data === `object`) {
+      e = { ...e, ...e.response.data }
+    }
+  }
```

Comment:
Is it normal to normalize 2 times? Can't we clean it up a little so we only assign once and please add some extra reasons why this is necessary. Maybe docs to the sdk 🤷. I know we'll be scratching our head about this in the future.

Reply:
I feel you. Was already painful to get all cases as error shapes are not documented 🤦

Reply:
Maybe it's good enough to add a comment to the top that we're dealing with legacy contentful sdk code, continue with caution 😛

Reply:
Should be a little bit cleaner now :)